flexget/plugins/sites/newznab.py

```diff
@@ -1,125 +1,615 @@
+                if year:
+                    list_append_unique(search_list, '%s (%s)' % (title, year), caseinsensitive=True)
+                list_append_unique(search_list, title, caseinsensitive=True)
+                if entry.get('search_strings'):
+                    list_combine_unique(search_list, entry('search_strings'), caseinsensitive=True)
 
-        if not len(rss.entries):
-            log.info('No results returned')
+        search_entry = Entry(entry)
+        for search_string in search_list:
+            search_string = search_string.strip()
+            if not search_string:
+                continue
+            search_entry['title'] = search_string
+            # update entry metadata
+            if show_lookup is True:
+                search_entry['series_name'] = search_string
+                for plugin_name in config['plugins_list']:
+                    if plugin_name == 'trakt_lookup':
+                        plugin.get_plugin_by_name(plugin_name).instance.lazy_series_lookup(search_entry)
+                        log.verbose('Doing `%s` for series: %s', plugin_name, search_string)
+                    elif plugin_name == 'tvmaze_lookup':
+                        plugin.get_plugin_by_name(plugin_name).instance.lazy_series_lookup(search_entry)
+                        log.verbose('Doing `%s` for series: %s', plugin_name, search_string)
+                    elif plugin_name == 'thetvdb_lookup':
+                        # TODO @Andy: do we need language support?
+                        plugin.get_plugin_by_name(plugin_name).instance.lazy_series_lookup(search_entry, 'en')
+                        log.verbose('Doing `%s` for series: %s', plugin_name, search_string)
+            elif show_lookup is False:
+                for plugin_name in config['plugins_list']:
+                    if plugin_name == 'trakt_lookup':
+                        plugin.get_plugin_by_name('trakt_lookup').instance.lazy_movie_lookup(search_entry)
+                        log.verbose('Doing `%s` for movie: %s', plugin_name, search_string)
+                    elif plugin_name == 'tmdb_lookup':
+                        plugin.get_plugin_by_name('tmdb_lookup').instance.lookup(search_entry)
+                        log.verbose('Doing `%s` for movie: %s', plugin_name, search_string)
+                    elif plugin_name == 'imdb_lookup':
+                        plugin.get_plugin_by_name('imdb_lookup').instance.lookup(search_entry)
+                        log.verbose('Doing `%s` for movie: %s', plugin_name, search_string)
+            if self._has_meta_id(search_entry) is not None:
+                search_entry['title'] = entry['title']  # keep org. title
+                return search_entry
+        return entry
 
-        for rss_entry in rss.entries:
+    def _build_base_url(self, entry, config):
+        # TODO @Andy: do 't=caps' check (many indexers wont correctly handle this, nzbhydra uses 'brute force' since 0.2.169)
```

Comment:
I understand that, but that's because they are annoying "TODO" or we would actually "do" them in the first place. I still think its better than nothing and is just a quick reminder of "ideas". So i don't like keeping a separated textfile or something more complex around per module. I'm well aware the most TODO will never done, yet i also read some TODO in other modules that actually helped understand what was still missing.